tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/strategy/ProgrammeBasedFilterStrategy.java

```diff
@@ -3,33 +3,80 @@
 import com.transformuk.hee.tis.security.model.Programme;
 import com.transformuk.hee.tis.security.model.UserProfile;
 import com.transformuk.hee.tis.security.util.TisSecurityHelper;
+import com.transformuk.hee.tis.tcs.api.enumeration.ProgrammeMembershipStatus;
+import com.transformuk.hee.tis.tcs.service.model.ColumnFilter;
+import java.util.List;
 import java.util.Optional;
 import java.util.Set;
+import java.util.stream.Collectors;
 import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.lucene.search.join.ScoreMode;
 import org.elasticsearch.common.collect.Tuple;
 import org.elasticsearch.index.query.BoolQueryBuilder;
 import org.elasticsearch.index.query.MatchQueryBuilder;
 import org.elasticsearch.index.query.NestedQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
 import org.springframework.stereotype.Component;
 
 @Component
 public class ProgrammeBasedFilterStrategy implements RoleBasedFilterStrategy {
 
-  private static final String COLUMN_FILTER = "programmeName";
+  private static final String COLUMN_FILTER = "programmeId";
```

Comment:
:+1:  This is ok given the rest of the code & my assumption it was `programmeName` to restrict user-selected filtering.  It would be useful to look at whether we can clear up some of the nesting & logic associated with avoiding multiple sets of filtering.  The previous commit for this line reads like a squash so it is hard to say.

I don't know why this was there despite having a "must" across filters. It is a bit of an awkward one.  I expect it was `programmeName` to restrict user-selected filtering.  It would be useful to look at whether we can clear up some of the nesting & logic associated with avoiding multiple sets of filtering.

Reply:
Yeah, I assumed it was `programmeName` for user-selected filtering too.

I'm not sure if I understand this comment properly. But the "must" filter makes sense to me as I think in this "must" filter, every filtering is required. 
This `programmeMembershipStatus` filter is very tricky. It can be used on its own, or on the user-selected `programmeName` column filter or role based `programmeId` filter. I found it was really hard to clean up any other bit of filtering.